### ISSUE TITLE
Admin interface must implement 3.1.2 Language of Parts for multilingual interfaces

### ISSUE THREAD

**Issue by thibaudcolas** [wagtail/wagtail#11376]
### Issue Summary

This is a problem for the whole Django / Python / gettext as far as I know – here’s how it manifests in Wagtail.

Wagtail supports multilingual admin interfaces, with "partial" language translations where an admin view will have content in multiple languages. For accessibility, it’s important that parts of the page in different languages are correctly marked up with the [`lang` attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/lang). This currently isn’t the case.

See: [WCAG SC 3.1.2: Language of Parts (Level AA)](https://www.w3.org/WAI/WCAG21/Understanding/language-of-parts.html)

### Steps to Reproduce

1. Go to https://static-wagtail-v5-2.netlify.app/admin-german/, or any other non-english variation of the CMS interface with english content
2. Note how some terms like "What's new in Wagtail 5.2" do not use the `lang` attribute, despite the lack of a translation

### Possible solutions

I have no idea how to approach this.

Related: #11375, #4694

### Working on this

Anyone can contribute to this, but we first need a solid plan to approach the task. View our [contributing guidelines](https://docs.wagtail.org/en/latest/contributing/index.html), add a comment to the issue once you’re ready to start.

**Comment by vanamaabhi2004** [comment]
![wangtail-bug](https://github.com/wagtail/wagtail/assets/126014969/4aa83ba4-1368-4df7-81d7-5fdeef78bc3b)



As above the tags can contain the lang attributes to the code .

**Comment by thibaudcolas** [comment]
@vanamaabhi2004 thanks for taking a look at this. What you’re screenshoting here is Wagtail content being in a different language, which isn’t what this issue is about. Please see #11375.

This issue is specifically about the Wagtail admin _interface_ being in multiple languages when someone uses it with a language we only have partial translations for.

**Comment by gasman** [comment]
@thibaudcolas I also can't see a good way of tackling this, short of totally reworking the way the `gettext` functions work - since those functions don't provide any way to identify when they've failed to find a translation and fallen back on the English text.

Is this something that actually needs addressing, though? For any given string that is missing a translation and not marked up with `lang="en"`, our preferred remedy would clearly be "add the translation" rather than "add the attribute". Obviously, I'm not suggesting that "just add 100% coverage of all languages and the problem will go away 😇 " is a realistic solution to this issue... just that we've always treated translation coverage as a work in progress that will never really be complete.

In the absence of a mechanism for inserting `lang="en"`, achieving full ATAG compliance for a given language will be conditional on having 100% translation coverage for that language. On the other hand, with such a mechanism in place we might technically tick a box on the checklist but I'm not sure that we can call the interface truly usable and accessible in that language. So, wouldn't it be reasonable to treat full cross-language ATAG compliance as the same work-in-progress effort - i.e. say "currently we can claim full ATAG compliance for the English, French, German, Dutch (...) versions of the Wagtail admin; for all other languages, we're working towards it and would welcome help"?

(On the other hand, if there's pressure on us to achieve full (technical) ATAG compliance across all languages on a shorter timescale than reaching 100% translation coverage, then I guess my argument is invalid and we do indeed need `lang="en"` as a stopgap.)

**Comment by thibaudcolas** [comment]
@gasman thank you for looking into this ☺️ Yeah, honestly short of a very creative `gettext` re-implementation I don’t really know whether/if/how we should tackle this.

This is actually a WCAG AA consideration ([SC 3.1.2: Language of Parts (Level AA)](https://www.w3.org/WAI/WCAG21/Understanding/language-of-parts.html)), so in some ways more pressing for us than ATAG. It’s only helpful for screen reader users though, and only changes how English words would be pronounced. Something we should certainly try to solve but for now I suspect even those people who could use it would welcome other improvements most.

For now I’ve reached out for help [on the web-a11y Slack](https://web-a11y.slack.com/archives/C071U5Y3H/p1705061813213989). The suggestion so far was "have gettext return HTML with the `lang` attribute in it", which wouldn’t be that simple in practice, and I’m not sure is a good idea with Django. This feels like a very broad "crowdsourced partial translations" problem so I’d hope we can learn from other projects on this – will have to think of other places where to raise this.

**Comment by riddhisharma-sudo** [comment]
I want to give it a try can you tell me the file location

**Comment by thibaudcolas** [comment]
@riddhisharma-sudo we don’t have any proposal on how to tackle this currently. At least in theory, the issue is with all of the parts of the admin interface which contain labels. If you want to work on this please review the problem first, make sure you fully understand it, and share possible solutions here.

**Comment by thibaudcolas** [comment]
I decided to take this to the Python forum in the hopes of identifying approaches that would work across other projects than Wagtail: [Accessibility of multilingual content with mixed translation](https://discuss.python.org/t/accessibility-of-multilingual-content-with-mixed-translation/89902). Not entirely sure where this will go but it’s worth a shot.
